SSA/Projects/InstCombine/Test.lean

```diff
@@ -383,3 +383,168 @@ def three_inst_macro_noreduc_proof (w : Nat) :
   simp_alive_meta
   simp_alive_ssa
   apply three_inst_stmt
+
+def one_inst_concrete_macro :=
+  [alive_icom ()|{
+  ^bb0(%arg0: i1):
+    %0 = "llvm.not" (%arg0) : (i1) -> (i1)
+    "llvm.return" (%0) : (i1) -> ()
+  }]
+
+set_option ssa.alive_icom_reduce false in
+def one_inst_concrete_macro_noreduce :=
+  [alive_icom ()|{
+  ^bb0(%arg0: i1):
+    %0 = "llvm.not" (%arg0) : (i1) -> (i1)
+    "llvm.return" (%0) : (i1) -> ()
+  }]
+
+def one_inst_concrete_com :
+  Com InstCombine.LLVM [InstCombine.Ty.bitvec 1] .pure (InstCombine.Ty.bitvec 1) :=
+  Com.lete (not 1 0) <|
+  Com.ret ⟨0, by simp [Ctxt.snoc]⟩
+
+def one_inst_concrete_stmt :
+  @BitVec.Refinement (BitVec 1) (LLVM.not e) (LLVM.not e) := by simp
+
+def one_inst_concrete_com_proof :
+  one_inst_concrete_com ⊑ one_inst_concrete_com := by
+  unfold one_inst_concrete_com
+  simp only [simp_llvm_wrap]
+  simp_alive_meta
```

Comment:
I noticed the proof state after `simp_alive_meta` still contains `ConcreteOrMVar.concrete`. If we change the normal form to be `ofNat`, we probably ought to change the relevant `instantiateFoo` functions to use `ofNat` instead of `.concrete`

Reply:
@alexkeizer Aha, then I'm happy to switch the normal form to `.concrete`. We should choose one, and I'd rather choose the one that aids consistency.

Reply:
@bollu, could you do the switching? I feel you are faster at it.

Reply:
Actually, I just did the switching. It was easy. This is now resolved.